hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterAdmin.java

```diff
@@ -103,11 +113,39 @@ public static void globalSetUp() throws Exception {
         createNamenodeReport("ns1", "nn1", HAServiceState.ACTIVE));
     stateStore.refreshCaches(true);
 
+    setUpMocks();
+  }
+
+  private static void setUpMocks() throws IOException {
     RouterRpcServer spyRpcServer =
         Mockito.spy(routerContext.getRouter().createRpcServer());
     Whitebox
         .setInternalState(routerContext.getRouter(), "rpcServer", spyRpcServer);
     Mockito.doReturn(null).when(spyRpcServer).getFileInfo(Mockito.anyString());
+
+    mockRpcClient = Mockito.spy(spyRpcServer.getRPCClient());
+    Whitebox
+        .setInternalState(spyRpcServer, "rpcClient", mockRpcClient);
+    RemoteLocation remoteLocation0 = new RemoteLocation("ns0", "/testdir", null);
+    RemoteLocation remoteLocation1 = new RemoteLocation("ns1", "/", null);
+    mockResponse0.put(remoteLocation0,
+        new HdfsFileStatus.Builder().build());
+    Mockito.doReturn(mockResponse0).when(mockRpcClient).invokeConcurrent(
+        Mockito.eq(Lists.newArrayList(remoteLocation0)),
+        Mockito.any(RemoteMethod.class),
+        Mockito.eq(false),
+        Mockito.eq(false),
+        Mockito.eq(HdfsFileStatus.class)
+    );
+    mockResponse1.put(remoteLocation1,
+        new HdfsFileStatus.Builder().build());
+    Mockito.doReturn(mockResponse1).when(mockRpcClient).invokeConcurrent(
```

Comment:
What are we doing with this? Is there a test actually using it?

Reply:
After adding the new logic in addMount and updateMount, the current tests would fail since they don't have the file created in destination ns. The mock is to avoid the failure for existing tests.
I can add a negative test.